jax/_src/lax/lax.py

```diff
@@ -3372,7 +3369,7 @@ def _transpose_batch_rule(batched_args, batch_dims, *, permutation):
 def _transpose_lower(ctx, x, *, permutation):
   aval_out, = ctx.avals_out
   if core.is_opaque_dtype(aval_out.dtype):
-    return aval_out.dtype._rules.transpose_mlir(ctx, x, permutation=permutation)
+    return aval_out.dtype._rules.transpose_mlir(ctx, aval_out, x, permutation=permutation)
```

Comment:
In general, we pass aval_out explicitly to the _rules methods because we should not assume that ctx.avals_out is the desired output aval for this particular op (it is the output aval for the currently running lowering rule).